src/Json.Schema.ToDotNet/ClassGenerator.cs

```diff
@@ -381,6 +402,95 @@ private ConstructorDeclarationSyntax GenerateDefaultConstructor()
                             SuffixedTypeName)));
         }
 
+        /// <summary>
+        /// Generates an initialization statement for each property for which the
+        /// schema specifies a default value.
+        /// </summary>
+        /// <remarks>
+        /// The resulting statements are inserted into the default constructor.
+        /// This ensures that the default values are set even if the object is
+        /// not the result of deserializing a JSON instance document.
+        /// <remarks>
+        /// <returns>
+        /// An array containing one initialization statement for each property
+        /// for which the schema specifies a default value.
+        /// </returns>
+        private ExpressionStatementSyntax[] GenerateDefaultInitializations()
+        {
+            var initializations = new List<ExpressionStatementSyntax>();
+
+            foreach (string propertyName in PropInfoDictionary.GetPropertyNames())
+            {
+                if (IncludeProperty(propertyName))
+                {
+                    PropertyInfo propInfo = PropInfoDictionary[propertyName];
+                    object defaultValue = propInfo.DefaultValue;
+
+                    ExpressionStatementSyntax initializationStatement = GenerateDefaultInitialization(propertyName, defaultValue);
+                    if (initializationStatement != null)
+                    {
+                        initializations.Add(GenerateDefaultInitialization(propertyName, defaultValue));
```

Comment:
>GenerateDefaultInitialization(propertyName, defaultValue [](start = 44, length = 56)

can't you just provide 'initializationStatement' here as an arg?

Reply:
Oh good heavens. I can't quite believe I did that! Fixed.

---
In reply to: [244416058](https://github.com/Microsoft/jschema/pull/60#discussion_r244416058) [](ancestors = 244416058)